### PR TITLE
Feat/update to flask 2

### PR DESCRIPTION
**Pull request by maudetes**
Fix https://github.com/etalab/data.gouv.fr/issues/1043

Changelogs for all lib upgraded have been checked.

### CODE REVIEW

File: udata/app.py [hunk 1]
```diff
@@ -78,6 +78,12 @@ def handle_http_exception(self, e):
         return super(UDataApp, self).handle_http_exception(e)
 
     def register_blueprint(self, blueprint, **kwargs):
+
+        if blueprint.name in self.blueprints:
+            # TODO: did the behaviour on multiple register_blueprint changed?
```
Comment (maudetes) [inline udata/app.py:83]: TODO: did the behaviour on multiple register_blueprint changed?

File: udata/core/dataset/api.py [hunk 2]
```diff
@@ -258,7 +258,7 @@ class DatasetRdfAPI(API):
     @api.doc('rdf_dataset')
     def get(self, dataset):
         format = RDF_EXTENSIONS[negociate_content()]
-        url = url_for('api.dataset_rdf_format', dataset=dataset.id, format=format)
+        url = url_for('api.dataset_rdf_format', dataset=dataset.id, format=format, _external=True)
```
Comment (maudetes) [inline udata/core/dataset/api.py:261]: TODO: did the behaviour on redirect location with external URL changed?

File: udata/core/user/models.py [hunk 3]
```diff
@@ -173,12 +174,16 @@ def followers_count(self):
         return self.metrics.get('followers', 0)
 
     def generate_api_key(self):
-        s = JSONWebSignatureSerializer(current_app.config['SECRET_KEY'])
-        byte_str = s.dumps({
+        # TODO: is this working?
```
Comment (maudetes) [inline udata/core/user/models.py:177]: TODO: validate that this is identical

File: udata/search/query.py [hunk 4]
```diff
@@ -72,5 +72,8 @@ def to_url(self, url=None, replace=False, **kwargs):
                     params[key] = value
         else:
             params['page'] = self.page
-        href = Href(url or request.base_url)
-        return href(params)
+        # Href has been deprecated and removed since https://github.com/pallets/werkzeug/pull/2085
```
Comment (maudetes) [inline udata/search/query.py:75]: TODO: check on a simple alternative to build an url with param :)

File: udata/tests/api/test_me_api.py [hunk 5]
```diff
@@ -286,6 +286,7 @@ def test_clear_apikey(self):
     def test_delete(self):
         '''It should delete the connected user'''
         user = self.login()
+        user_email = user.email
```
Comment (maudetes) [inline udata/tests/api/test_me_api.py:289]: TODO: Why does `user.email` get updated now without reload and not before?

File: requirements/install.in [hunk 6]
```diff
@@ -7,38 +7,36 @@ blinker==1.4
...
+Flask-BabelEx==0.9.4
+Flask-Caching==2.0.2
 Flask-CDN==1.5.3
 Flask-Cors==3.0.8
-flask-fs==0.6.1
+flask-storage==1.0.0
 Flask-Gravatar==0.5.0
-Flask-Login==0.4.1
+Flask-Login==0.6.2  # an issue with 0.6.2? https://github.com/maxcountryman/flask-login/issues/729
 Flask-Mail==0.9.1
```
[the remark this answers is not in the document]
Comment (maudetes) [reply requirements/install.in:21]: Disclaimer: it didn't work due to lazystring not being pickable anymore when serializing for celery. Tried to update Flask-BabelEx without success at first try: https://github.com/etalab/data.gouv.fr/issues/1059.
Fixed the issue locally with https://github.com/opendatateam/udata/commit/a438bcf9a6f48decb1ec422d0e8cdaca0d7d590d.

**Review by maudetes — COMMENTED** [review]
Work in Progress

File: udata/commands/__init__.py [hunk 7]
```diff
@@ -168,8 +168,9 @@ def init_logging(app):
     return app
 
 
-def create_cli_app(info):
-    app = create_app(info.settings, init_logging=init_logging)
+def create_cli_app():
+    # TODO: how can we get info settings?
```
[the remark this answers is not in the document]
Comment (maudetes) [reply udata/commands/__init__.py]: Done in https://github.com/opendatateam/udata/pull/2816/commits/2e1db19cf35dfbdd83a686698138f131eed9df8c. See https://github.com/pallets/flask/issues/3552

File: udata/tests/api/test_me_api.py (hunk 5 above)
Comment (maudetes) [reply udata/tests/api/test_me_api.py:289]: Doesn't seem to have any side effects.

File: requirements/install.in [hunk 8]
```diff
@@ -7,38 +7,36 @@ blinker==1.4
...
+Jinja2==3.1.2
 jsonschema==3.2.0
 kombu[redis]==4.4.0
 lxml==4.4.2
 mistune==0.8.4
 mongoengine==0.20.0
 msgpack==0.6.2
 netaddr==0.7.19
-pillow==8.0.0
+pillow==9.2.0
```
Comment (maudetes) [inline requirements/install.in:39]: Is it used only for tests? Or am I missing something?

**Review by quaxsze — APPROVED** [review]

**Pull request merged**